extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcRecorder.java

```diff
@@ -43,15 +43,24 @@
         return () -> new DefaultTokenIntrospectionUserInfoCache(config, vertx.get());
     }
 
-    public Supplier<TenantConfigBean> setup(OidcConfig config, Supplier<Vertx> vertx, TlsConfig tlsConfig) {
+    public Supplier<TenantConfigBean> setup(OidcConfig config, Supplier<Vertx> vertx, TlsConfig tlsConfig,
+            Map<String, Supplier<Optional<OidcTenantConfig>>> knownProviders) {
         final Vertx vertxValue = vertx.get();
 
         String defaultTenantId = config.defaultTenant.getTenantId().orElse(DEFAULT_TENANT_ID);
         TenantConfigContext defaultTenantContext = createStaticTenantContext(vertxValue, config.defaultTenant, tlsConfig,
                 defaultTenantId);
 
         Map<String, TenantConfigContext> staticTenantsConfig = new HashMap<>();
-        for (Map.Entry<String, OidcTenantConfig> tenant : config.namedTenants.entrySet()) {
+        Map<String, OidcTenantConfig> tenants = new HashMap<>(config.namedTenants);
+        for (var e : knownProviders.entrySet()) {
+            var provider = e.getValue().get();
+            if (provider.isPresent()) {
+                tenants.put(e.getKey(), provider.get());
```

Comment:
Sorry, but it is not correct - it creates a dedicated tenant per every social provider config, but it won't allow to override a specific property using `OidcTenantConfig` - with this approach we'd end up moving a good portion of `OidcTenantConfig` properties under the `github` and other provider specific blocks. I'd say it is a problem

But also equally important, it does not allow to vary with the provider, each OIDC tenant allows to have:
```
quarkus.oidc.github1.provider=github
quarkus.oidc.github1.authentication.scopes=a

quarkus.oidc.github2.provider=github
quarkus.oidc.github2.authentication.scopes=b
```
It may not be something one would want to do in production but for tests it is very useful.

Reply:
Yeah, so continuing from the earlier comment, so you have `Map<String, Supplier<Optional<OidcTenantConfig>>> knownProviders`.
But instead of creating tenants as you have typed here, the following will be more aligned with the OIDC tenancy:
```
if (oidcTenantConfig.provider.isPresent()) {
    OidcTenantConfig thisIsCurrentTenantConfig = new OidcTenantConfig(knownProviders.get(oidcTenantConfig.provider.get()));
    // continue dealing with this tenant config
}
```